lib/google_analytics_adapter.rb

```diff
@@ -3,20 +3,38 @@ def initialize(visit)
     @visit = visit
   end
 
+  def to_json
+    {
+      prison_name: prison_name,
+      number_of_slots: number_of_slots,
+      prisoner_age: prisoner_age,
+      visitor_age: visitor_age,
+      number_of_adult_visitors: number_of_adult_visitors,
+      number_of_child_visitors: number_of_child_visitors,
+      slot_times: slot_times,
+      slot_weekdays: slot_weekdays,
+      days_to_first_slot: days_to_first_slot,
+      days_to_last_slot: days_to_last_slot,
+      completed_at: Time.now.utc.strftime("%Y%m%d%H%M%S")
+    }.to_json
+  end
+
+  private
+
   def prison_name
     @visit.prisoner.prison_name
   end
 
   def number_of_slots
-    nonempty_slots.count
+    date_sorted_slots.count
   end
 
   def slot_times
-    nonempty_slots.map { |s| s.times }
+    date_sorted_slots.map { |s| s.times }
```

Comment:
We actually did it, but then we thought is not a change due to rubocop so we were thinking to create a new branch and add those changes!